src/components/data/DataTable.js

```diff
@@ -114,6 +115,17 @@ const DataTable = props => {
           { billingProject: googleProject, dataReference: snapshotName } :
           { filterTerms: activeTextFilter })
       }))
+    // find all the unique attribute names contained in the current page of results
+    const allAttrNames = _.uniq(_.flatMap(_.keys, _.map('attributes', results)))
+    // add any attribute names from the current page of results to those found in metadata.
+    // This allows for the stale metadata (e.g. the metadata cache is out of date).
+    // For the time being, the uniquness check MUST be case-insensitive (e.g. { sensitivity: 'accent' })
+    // in order to prevent case-divergent columns from being displayed, as that would expose some other bugs
+    const newAttrsForThisType = _.uniqWith((left, right) => !left.localeCompare(right, undefined, { sensitivity: 'accent' }),
+      _.concat(entityMetadata[entityType].attributeNames, allAttrNames)).sort()
+    const newMetadata = _.cloneDeep(entityMetadata)
+    newMetadata[entityType].attributeNames = newAttrsForThisType
+    setEntityMetadata(newMetadata)
```

Comment:
question for the UI experts: do I need to explicitly call `setEntityMetadata()` to persist my changes to state, or could I just use:
```entityMetadata[entityType].attributeNames = newAttrsForThisType```
both seem to function the same as I click through the app, but I don't know if there is an underlying difference.